pkg/arvo/app/chat.hoon

```diff
@@ -225,109 +146,70 @@
     (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
   ?+  i.wir
     (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
-    ::
-    ::  %circles wire
-    ::
+  ::
       %circles
     ?>  ?=(%circles -.piz)
-      =/  str  %=  str.sta
-        circles  cis.piz
-      ==
-      :-  (send-chat-update [[%circles cis.piz] str])
-      this(str.sta str)
+    =.  str  str(circles cis.piz)
+    :_  this(str str)
+    (send-chat-update [[%circles cis.piz] str])
+  ::
+      %circle
+    ?>  ?=(%circle -.piz)
+    =/  circle=circle:hall  [our.bol &3:wir]
+    ?.  =(circle [our.bol %inbox])
     ::
-    ::  %circle wire
+    ::  fill remote configs with message data
     ::
-      %circle
-::      ::
-::      ::  %circle prize
-::      ::
-::        %circle
-      ?>  ?=(%circle -.piz)
-      =/  circle/circle:hall  [our.bol &3:wir]
-      ?:  =(circle [our.bol %inbox])
+      =*  messages  messages.str
+      =/  circle=circle:hall  [`@p`(slav %p &2:wir) &3:wir]
+      =/  peers=(map circle:hall (set @p))
+        %-  ~(rep by rem.pes.piz)
+        |=  [[cir=circle:hall grp=group:hall] acc=(map circle:hall (set @p))]
+        ^+  acc
+        (~(put by acc) cir (silt (turn ~(tap by grp) head)))
+      ::
+      =/  sta
+        %=  str
+          messages  (~(put by messages) circle nes.piz)
+          peers
+            %-  ~(uni by peers.str)
+            (~(put by peers) circle ~(key by loc.pes.piz))
+        ==
+      :_  this(str sta)
+      (send-chat-update [[%messages circle 0 (lent messages) nes.piz] sta])
     ::
     ::  fill inbox config and remote configs with prize data
     ::
-        =/  configs
-          %-  ~(uni in configs.str.sta)
+    =/  circles=(list circle:hall)  (turn ~(tap in src.loc.cos.piz) head)
+    ::
+    =/  peers=(map circle:hall (set @p))
+      %-  ~(rep by rem.pes.piz)
+      |=  [[cir=circle:hall grp=group:hall] acc=(map circle:hall (set @p))]
+      ^+  acc
+      (~(put by acc) cir (silt (turn ~(tap by grp) head)))
+    ::
+    :-
+      %+  turn  ~(tap in (~(del in (silt circles)) [our.bol %inbox]))
```

Comment:
I would maybe split out the `del in`, this is a little cramped.

Reply:
I think this is fine, really. Otherwise more inline variable declarations. :^)

Reply:
```hoon
%+  turn
  %~  tap  in
  %-  ~(del in (silt circles))
  [our.bol %indox]
rest-of-the-code
```

?